src/category_theory/abelian/projective.lean

```diff
@@ -37,4 +40,106 @@ lemma exact_d_f {X Y : C} (f : X ⟶ Y) : exact (d f) f :=
 
 end
 
+namespace ProjectiveResolution
+
+/- We have to jump through some hoops to get `ProjectiveResolution.of` to typecheck! -/
```

Comment:
Would it be possible to add a bit more comments about the (mathematical) role of `B'`, `X'`, etc... in the construction. The one-line summary is clear: repeatedly pick a projective cover. But exploding this one-liner into Lean seems to require some fairly non-trivial hacking, and it would be good to have a clear description of how the components fit together.